src/battle_script_commands.c

```diff
@@ -16524,6 +16525,22 @@ void BS_TryDefog(void)
     }
 }
 
+void BS_TryUpperHand(void)
+{
+	NATIVE_ARGS(const u8 *failInstr);
+	
+	if (gProtectStructs[gBattlerTarget].obstructed) // Fails if target is obstructed
```

Comment:
Don't use an if else if chain if the result is the same for each branch. Just use `||`